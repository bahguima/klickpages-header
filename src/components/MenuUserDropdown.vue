<template>
<div class="dropdown menu-user">
  <a href="javascript:;" class="kp-user dropdown-toggle clearfix" data-toggle="dropdown">
    <span class="btn-user">{{topBarConfig.user.name[0]}}</span>
    <span class="user__text">{{topBarConfig.user.name}}</span>
    <span class="btn">
      <i class="fa fa-angle-down"></i>
    </span>
  </a>
  <ul class="dropdown-menu">
    <li>
      <a class="option" :href="topBarConfig.profile.url">
        <profile-icon />
        {{topBarConfig.profile.title}}
      </a>
    </li>
    <li>
      <a class="option" :href="topBarConfig.mailservice.url">
        <mail-service-icon />
        {{topBarConfig.mailservice.title}}
      </a>
    </li>
    <li>
      <a class="option" :href="topBarConfig.domain.url">
        <domain-icon />
        {{topBarConfig.domain.title}}
      </a>
    </li>
    <li>
      <a class="option" :href="topBarConfig.logout.url">
        <logout-icon />
        {{topBarConfig.logout.title}}
      </a>
    </li>
  </ul>
</div>
</template>

<script>
import { mapGetters } from 'vuex';
import DomainIcon from './icons/Domain.vue';
import MailServiceIcon from './icons/MailService.vue';
import ProfileIcon from './icons/Profile.vue';
import LogoutIcon from './icons/Logout.vue';
export default {
  computed: mapGetters({
    topBarConfig: 'topBar/config',
  }),
  components: {
    DomainIcon,
    ProfileIcon,
    MailServiceIcon,
    LogoutIcon,
  },
};
</script>
