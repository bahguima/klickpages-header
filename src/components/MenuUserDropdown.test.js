import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import MenuUserDropdown from './MenuUserDropdown.vue';

describe('components/MenuUserDropdown.vue', () => {
  const localVue = createLocalVue();
  localVue.use(Vuex);

  const topBarConfig = {
    user: {
      name: 'Foo Bar',
    },
    profile: {
      url: 'profile-url.com',
      title: 'profile',
    },
    domain: {
      url: 'domain-url.com',
      title: 'domain',
    },
    logout: {
      url: 'logout-url.com',
      title: 'logout',
    },
  };

  const topBar = {
    namespaced: true,
    getters: {
      config: () => topBarConfig,
    },
  };

  let wrapper;

  describe('when component is mounted', () => {
    beforeAll(() => {
      const store = new Vuex.Store({ modules: { topBar } });
      wrapper = shallowMount(MenuUserDropdown, { localVue, store });
    });

    afterAll(() => {
      wrapper.destroy();
      jest.clearAllMocks();
    });

    it('should render the html correctly', () => {
      expect(wrapper.element).toMatchSnapshot();
    });

    it('should render the first letter of the name on the user button', () => {
      const userButton = wrapper.element.getElementsByClassName('btn-user')[0];
      expect(userButton.innerHTML).toEqual(topBarConfig.user.name[0]);
    });

    it('should render the user name in the menu', () => {
      const userNameSpan = wrapper.element.getElementsByClassName('user__text')[0];
      expect(userNameSpan.innerHTML).toEqual(topBarConfig.user.name);
    });
  });
});
