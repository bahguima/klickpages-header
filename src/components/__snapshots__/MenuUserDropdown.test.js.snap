// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/MenuUserDropdown.vue when component is mounted should render the html correctly 1`] = `
<div
  class="dropdown menu-user"
>
  <a
    class="kp-user dropdown-toggle clearfix"
    data-toggle="dropdown"
    href="javascript:;"
  >
    <span
      class="btn-user"
    >
      F
    </span>
     
    <span
      class="user__text"
    >
      Foo Bar
    </span>
     
    <span
      class="btn"
    >
      <i
        class="fa fa-angle-down"
      />
    </span>
  </a>
   
  <ul
    class="dropdown-menu"
  >
    <li>
      <a
        class="option"
        href="profile-url.com"
      >
        <profile-icon-stub />
        
        profile
      
      </a>
    </li>
     
    <li>
      <a
        class="option"
        href="domain-url.com"
      >
        <domain-icon-stub />
        
        domain
      
      </a>
    </li>
     
    <li>
      <a
        class="option"
        href="logout-url.com"
      >
        <logout-icon-stub />
        
        logout
      
      </a>
    </li>
  </ul>
</div>
`;
